source/algorithms.tex

```diff
@@ -1126,39 +1126,39 @@
 or
 \tcode{Input\-Iterator2},
 the template argument shall satisfy the
-requirements of an input iterator\iref{input.iterators}.
+\textit{Cpp98InputIterator} requirements\iref{input.iterators}.
 \item
 If an algorithm's template parameter is named
 \tcode{OutputIterator},
 \tcode{OutputIterator1},
 or
 \tcode{Output\-Iterator2},
-the template argument shall satisfy the requirements
-of an output iterator\iref{output.iterators}.
+the template argument shall satisfy the
+\textit{Cpp98OutputIterator} requirements\iref{output.iterators}.
 \item
 If an algorithm's template parameter is named
 \tcode{ForwardIterator},
 \tcode{ForwardIterator1},
 or
 \tcode{Forward\-Iterator2},
-the template argument shall satisfy the requirements
-of a forward iterator\iref{forward.iterators}.
+the template argument shall satisfy the
+\textit{Cpp98ForwardIterator} requirements\iref{forward.iterators}.
```

Comment:
The issue here is that a reference to "forward iterator" (plain text) refers to the entire section [forward.iterators], including (for example) the multi-pass guarantee. In contrast, _Cpp98ForwardIterator_ appears to refer to the table only (which just gives a bit of syntax). That feels like a regression to me.

Reply:
"forward iterators" will be ambiguous once Ranges are merged. I understand your point about the distinction between the table and the additional requirements outside the table, though. Do you agree it would be clearer if I retitle the existing sections e.g. "Cpp98ForwardIterator requirements"?

Reply:
Yes, I think so. I fear there are Cpp98 concepts definitions that don't have a section of their own, though, so this strategy might fail here and there. Before you subscribe for more work, @zygoloid should really chime in how he wants to proceed here.
In any case, make sure all those changes are a (one) separate commit so that they can be messed with easily.